src/api/hooks/useGetANS.ts

```diff
@@ -54,11 +54,21 @@ function getFetchAddressUrl(network: NetworkName, name: string) {
   return `https://www.aptosnames.com/api/${network}/v1/address/${name}`;
 }
 
+function truncateAptSuffix(name: string): string {
```

Comment:
thanks for working on this issue! wondering can we use regex instead to find all the matching scenarios where "cathy.", "cathy.a", "cathy.ap", "cathy.apt" are all qualified and can be translated to "cathy"? 

some regex like `const regex = /^([a-zA-Z0-9-]+)(\.apt|\.ap|\.a|\.?)$/;`

Reply:
i feel like if we introduce more characters to ANS in the future, the regex will be more and more complex? generally i tried to avoid using regex because of it's difficult to read or test. thoughts?

Reply:
in the forseeable future, we have primary names which is like "cathy.apt" and subdomains, which is like  "cathy.sub.apt". right now we only display primary names, and regex won't get complex as long as it fulfills the requirement "Name should have between 3 to 63 characters and contain only: lowercase (a-z), numbers (0-9), hyphen (-). A name may not start or end with a hyphen", which ANS uses regex to check against input string's validity. 

let me know if you have a strong preference here! we can discuss async as well

Reply:
This is the regex we widely use 

 /^[a-z\d][a-z\d-]{1,61}[a-z\d]$/;

I think it is somewhere is the ANS codebase, @angieyth might be nice to export it as a util function from the sdk - something like isNameValid